test/test_framework.py

```diff
@@ -750,6 +750,26 @@ def test_ban_concurrent_frameworks(self):
         self.assertIn('database is locked', str(cm.exception))
         f.close()
 
+    def test_snapshot_saving_restricted_to_simple_types(self):
+        # this can not be saved, as it has not simple types!
+        to_be_saved = {"bar": TestFramework}
+
+        class FooEvent(EventsBase):
+            def snapshot(self):
+                return to_be_saved
+
+        handle = Handle(None, "a_foo", "some_key")
+        event = FooEvent()
+
+        framework = self.create_framework()
+        framework.register_type(FooEvent, None, handle.kind)
+        with self.assertRaises(ValueError) as cm:
+            framework.save_snapshot(event)
+        expected = (
+            "Can not save the data for FooEvent, must contain only simple types: "
```

Comment:
To bikeshed on the message a bit:

"unable to save the data for FooEvent, it must contain only simple types: "

I think that is slightly better. I also think there was a design decision early to have error messages start lower case instead of Sentence case. (eg: raise ModelError('cannot set a pod spec as this unit is not a leader')).

I believe this is so they compose better if you have wrappers that catch the error and then add more context.